app/services/service_service.js

```diff
@@ -51,14 +52,14 @@ function updateServiceName (serviceExternalId, serviceName, correlationId) {
     .then(result => {
       const gatewayAccountIds = lodash.get(result, 'gateway_account_ids', [])
       // Update gateway account service names
-      if (gatewayAccountIds.length > 0) {
-        return q.all([].concat(gatewayAccountIds).map(gatewayAccountId => {
-          connectorClient().patchServiceName(gatewayAccountId, serviceName, correlationId)
-        })).then(() => {
-          q.resolve(new Service(result))
-        })
-      } else {
+      if (gatewayAccountIds.length <= 0) {
         return q.resolve(new Service(result))
+      } else {
+        return q.all([
+          ...gatewayAccountIds.map(gatewayAccountId => connectorClient.patchServiceName(gatewayAccountId, serviceName, correlationId)),
+          ...gatewayAccountIds.map(gatewayAccountId => productsClient.product.updateServiceNameOfProductsByGatewayAccountId(gatewayAccountId, serviceName))
```

Comment:
no need to pass the `correlationId` here ?

Reply:
nope. productsclient gets it by other means.

Reply:
😳  👍